libminifi/include/core/Processor.h

```diff
@@ -180,25 +180,24 @@ class Processor : public Connectable, public ConfigurableComponent, public state
 
   bool addConnection(Connectable* connection);
 
-  virtual void onTrigger(const std::shared_ptr<ProcessContext> &context, const std::shared_ptr<ProcessSessionFactory> &sessionFactory);
-  void onTrigger(ProcessContext *context, ProcessSessionFactory *sessionFactory);
-
   bool canEdit() override {
     return !isRunning();
   }
 
-  virtual void onTrigger(const std::shared_ptr<ProcessContext> &context, const std::shared_ptr<ProcessSession> &session) {
-    onTrigger(context.get(), session.get());
-  }
-  virtual void onTrigger(ProcessContext* /*context*/, ProcessSession* /*session*/) {
-  }
   void initialize() override {
   }
-  virtual void onSchedule(const std::shared_ptr<ProcessContext> &context, const std::shared_ptr<ProcessSessionFactory> &sessionFactory) {
-    onSchedule(context.get(), sessionFactory.get());
+
+  virtual void onTrigger(const std::shared_ptr<ProcessContext>& context, const std::shared_ptr<ProcessSessionFactory>& session_factory);
+
+  virtual void onTrigger_2(const std::shared_ptr<ProcessContext>& context, const std::shared_ptr<ProcessSession>& session) {
```

Comment:
I couldnt come up with a better name yet, I've changed it from onTrigger so its easier to find the remaining occurences where we rely on shared_ptr-s

We could either revert this back to onTrigger or come up with a slightly better name?